File: src/grpc/convert/wkt.ts
import * as wkt_timestamp from 'google-protobuf/google/protobuf/timestamp_pb';
import type { Timestamp } from '../../entity'

export function timestampProto(v: Timestamp): wkt_timestamp.Timestamp {
  const ret = new wkt_timestamp.Timestamp();
  ret.setSeconds(v.seconds)
  ret.setNanos(v.nanos)
  return ret
}

export function timestampFromProto(v: wkt_timestamp.Timestamp): Timestamp {
  return v.toObject() as Timestamp
}

File: pkg/grpc/message/wkt.go
package message

import (
	"time"

	"github.com/gogo/protobuf/types"
)

// TimestampFromProto convert *types.Timestamp to time.Time
func TimestampFromProto(value *types.Timestamp) (ret time.Time) {
	ret, _ = types.TimestampFromProto(value)
	return ret
}

// TimestampProto convert time.Time to *types.Timestamp
func TimestampProto(value time.Time) (ret *types.Timestamp) {
	ret, _ = types.TimestampProto(value)
	return ret
}
